smithy-build/src/main/java/software/amazon/smithy/build/SmithyBuildImpl.java

```diff
@@ -48,7 +49,16 @@
 
 final class SmithyBuildImpl {
     private static final Logger LOGGER = Logger.getLogger(SmithyBuild.class.getName());
-    private static final Pattern PATTERN = Pattern.compile("^[A-Za-z0-9\\-_.]+$");
+
+    // Must start with one or more alphanumerics, followed by zero or more alphanumerics, "-", "_", or ".".
+    private static final String PATTERN_PART = "[A-Za-z0-9]+[A-Za-z0-9\\-_.]*";
```

Comment:
We also need to update the pattern in [the docs](https://smithy.io/2.0/guides/building-models/build-config.html#build-info-plugin).

> Plugin names MUST match the following pattern: `^[A-Za-z0-9\-_.]+$.`